### ISSUE TITLE
[1.2.0] Z-wave battery devices (motion sensors) queried by Homey for security info while asleep, blocking communication

### ISSUE THREAD

**Issue by TedTolboom** [athombv/homey#1468]
Follow-up of #1442 (2nd instance mentioned), on request of @jeroenvollenbrock split into separate issue; also since multiple users report the same issue

**What is your Homey version (Settings → System)?**
1.2.0 (latest non-RC experimental release)

**What did you try to achieve?**
Fibaro motion sensor-plus to report motion instantly

**What is the result**
Delayed response / no response at all from the motion sensors
In some cases, manually waking up the motion sensor solves the issue / clears the queue of requests

**Observations**
Out of the Z-wave log, it appears that Homey is querying the motion sensor for it's security key while the sensor is not awake...

Section out of the [170412_Z-wave log.txt](https://github.com/athombv/homey/files/917908/170412_Z-wave.log.txt); node 19 and 20 are the Fibaro Motion sensors:
```
[2017-04-12T21:17:10.507Z] Node[20]: Received application command for COMMAND_CLASS_SECURITY, data: 0x40
[2017-04-12T21:17:10.509Z] Node[20]: sendData to COMMAND_CLASS_SECURITY, params 0x80b23f098ea3ab608a
[2017-04-12T21:17:10.511Z] Command start: sendData
[2017-04-12T21:17:10.547Z] ProcessSendData[512]: To node: 20 with data: 0x9880b23f098ea3ab608a and txOptions: ACK,AUTO_ROUTE
[2017-04-12T21:17:10.870Z] Node[20]: Received application command for COMMAND_CLASS_SECURITY, data: 0x40
[2017-04-12T21:17:10.875Z] Node[20]: sendData to COMMAND_CLASS_SECURITY, params 0x80249dd92081f1500f
[2017-04-12T21:17:10.875Z] Command start: sendData
[2017-04-12T21:17:10.995Z] Node[20]: Received application command for COMMAND_CLASS_SECURITY, data: 0x40
[2017-04-12T21:17:10.996Z] Node[20]: sendData to COMMAND_CLASS_SECURITY, params 0x8018005350d7f57d83
[2017-04-12T21:17:10.996Z] Command start: sendData
[2017-04-12T21:17:11.979Z] Node[20]: Received application command for COMMAND_CLASS_SECURITY, data: 0x8117e692ca890e5d2fd85457e6b858661bdc3cb227bfec179dd3344c
[2017-04-12T21:17:12.006Z] Node[20]: Decapsulated frame from COMMAND_CLASS_SECURITY to COMMAND_CLASS_NOTIFICATION, data 0x05000000ff070800
[2017-04-12T21:17:12.542Z] Node[8]: sendData to COMMAND_CLASS_SWITCH_MULTILEVEL, params 0x014f05
[2017-04-12T21:17:12.543Z] Node[8]: sendData to COMMAND_CLASS_SECURITY, params 0x40
[2017-04-12T21:17:12.543Z] Command start: sendData
[2017-04-12T21:17:14.092Z] ProcessSendData[512]: Error: TRANSMIT_COMPLETE_NO_ACK
[2017-04-12T21:17:14.097Z] ProcessSendData[515]: To node: 8 with data: 0x9840 and txOptions: ACK,AUTO_ROUTE,EXPLORE
[2017-04-12T21:17:14.103Z] Command end: sendData
[2017-04-12T21:17:14.242Z] ProcessSendData[514]: To node: 20 with data: 0x988018005350d7f57d83 and txOptions: ACK,AUTO_ROUTE
[2017-04-12T21:17:14.247Z] Command end: sendData
```
Possible that a healing function out of the core is initiating these triggers, but does not differentiate between permanent powered and most of the time sleeping, battery powered devices.

**previous log as part of #1442**
170401: Inability to control any Z-wave devices
Trying to switch of my lights yesterday evening, none of the devices responded to the Homey flows (devices were responsive with direct association).

I noticed several security commands send to my battery powered Fibaro Motion Sensor Plus devices (node 19 and 20)...

Z-wave log:
[170401_Z-wave log.txt](https://github.com/athombv/homey/files/888237/170401_Z-wave.log.txt)

manually waking up these devices cleared the lockup and devices became responsive again (never experienced this one pre-RC10)

**Comment by jeroenbos22** [comment]
I also have this

**Comment by jeroenvollenbrock** [comment]
Could you please also send a Homey log?

**Comment by robiebab** [comment]
still active on 1.2.1 status from node 85 is not deliverd for some reason. mostion is still active. 7C59E2772B i don't know when the status whas triggerd, only that is still hasnt been updated. if i walk by now then the status will get updated.

**Comment by caseda** [comment]
just digging trough a small 500 line log.. (all from 1 device)
noticed a very familiar/repeated patern
but a small recap explains more:

71x (out of the 500) homey is being asked for a security key!
71x homey is honing to all those questions (but not receiving a ACK back)

in a timespan of 1 hour,
where there are 3 big timeouts of 12-15 minutes
and one of 6 minutes
(homey slow downs?)

the rest is all in fast "succession" 0 - 2 seconds
with every 5-6 times a jump to +/-40 seconds (normal device timeouts)

is there supposed to be an ACK signal send with the security key, but there is not?
or the security key failing to send at all.

**Comment by jeroenvano** [comment]
I am experiencing the same issues, also with the FMS ZW+ devices, see https://forum.athom.com/discussion/comment/51357/#Comment_51357

I can share the log if needed

**Comment by WeeJeWel** [comment]
Is this still an issue on 1.3.0-rc.7?

**Comment by jeroenbos22** [comment]
Yes after a couple of days had it again once now, jeroenvollenbrock told me sensor was not responding but restart of homey fixed it without touching the sensor

**Comment by TedTolboom** [comment]
Still occuring on 1.3.0-RC8: still see a lot of senddata_timeouts in combination with the Fibaro motion sensors and COMMAND_CLASS_SECURITY related logs:
```
[2017-05-22T08:34:04.055Z] ProcessSendData[18822]: To node: 20 with data: 0x98806d3a7f9a430c6708 and txOptions: ACK,AUTO_ROUTE
[2017-05-22T08:34:06.107Z] ProcessSendData[18822]: Error: senddata_timeout
```
Node 19 and 20 are the Fibaro motion sensors (PLUS version)
[170522_Z-wave log.txt](https://github.com/athombv/homey/files/1018200/170522_Z-wave.log.txt)

**Comment by satbopper** [comment]
i can confirm this,in my case it becomes from an aeotec multisensor 6
Command[677] start: sendData
[2017-05-22T08:43:53.255Z] ProcessSendData[668]: To node: 78 with data: 0x8408 and txOptions: ACK,AUTO_ROUTE,EXPLORE
[2017-05-22T08:43:55.273Z] ProcessSendData[668]: Error: senddata_timeout

**Comment by WeeJeWel** [comment]
Thanks for the logs, I will investigate why this happens.

**Comment by philsniff** [comment]
log: 8FEEA3093B

hey @WeeJeWel , i have the first time some "senddata_timeout" with my fibaro double switch 2 with 1.3.0 rc9!

[2017-05-30T20:51:23.069Z] Node[32]: sendData to COMMAND_CLASS_SWITCH_BINARY, params 0x01ff
[2017-05-30T20:51:23.076Z] Node[32]: sendData to COMMAND_CLASS_SECURITY, params 0x40
[2017-05-30T20:51:23.078Z] Command[67] start: sendData
[2017-05-30T20:51:23.087Z] ProcessSendData[65]: To node: 32 with data: 0x9840 and txOptions: ACK,AUTO_ROUTE
[2017-05-30T20:51:25.101Z] ProcessSendData[65]: Error: senddata_timeout
[2017-05-30T20:51:25.107Z] Command[67] end: sendData
[2017-05-30T20:51:28.376Z] Node[32]: sendData to COMMAND_CLASS_SWITCH_BINARY, params 0x0100
[2017-05-30T20:51:28.382Z] Node[32]: sendData to COMMAND_CLASS_SECURITY, params 0x40
[2017-05-30T20:51:28.385Z] Command[68] start: sendData
[2017-05-30T20:51:28.392Z] ProcessSendData[66]: To node: 32 with data: 0x9840 and txOptions: ACK,AUTO_ROUTE
[2017-05-30T20:51:30.412Z] ProcessSendData[66]: Error: senddata_timeout
[2017-05-30T20:51:30.419Z] Command[68] end: sendData
[2017-05-30T20:51:37.595Z] Command[69] start: getNetworkTopology
[2017-05-30T20:51:37.960Z] Command[69] end: getNetworkTopology
[2017-05-30T20:55:17.849Z] Command[70] start: getNetworkTopology
[2017-05-30T20:55:17.849Z] Command[70] start: getNetworkTopology
[2017-05-30T20:55:18.718Z] Command[70] end: getNetworkTopology

before rc9 it worked for 3 days (i included the double switch on saturday 27.05.) with 1.3.0 rc7.

**Comment by DeefK** [comment]
Here with a fibaro motion sensor:

[2017-06-13T17:57:29.129Z] Node[13]: Received application command for COMMAND_CLASS_SECURITY, data: 0x81242722bb6048dcc86b35d75276bc4bf1363c0a45e3819e72c595e2
[2017-06-13T17:57:29.172Z] Node[13]: Decapsulated frame from COMMAND_CLASS_SECURITY to COMMAND_CLASS_NOTIFICATION, data 0x05000000ff070800
[2017-06-13T17:57:29.219Z] ProcessSendData[37]: To node: 13 with data: 0x98800b652dd69ff2d514 and txOptions: ACK,AUTO_ROUTE
[2017-06-13T17:57:29.243Z] Command[37] end: sendData
[2017-06-13T17:57:31.282Z] ProcessSendData[37]: Error: senddata_timeout
[2017-06-13T17:57:31.294Z] Command[36] end: sendData
